frontend/src/views/Create.tsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import { useHistory } from 'react-router-dom';
+import { ENTER_NICKNAME_PAGE, EnterNicknamePage } from '../components/core/EnterNickname';
+import { errorHandler } from '../api/Error';
+import { createRoom, Room, RoomParams } from '../api/Room';
+
+function CreateGamePage() {
+  // Get history object to be able to move between different pages
+  const history = useHistory();
+
+  // Creates a room with the user as the host, and joins that same waiting room.
+  const createJoinWaitingRoom = (nickname: string) => new Promise<undefined>((resolve, reject) => {
```

Comment:
The reason why I make Promises for both of these is because I need to know when to call `setLoading(false)`. If this wasn't a promise, and instead simply a method, then I would either just never call `setLoading(false)` (which seems strange), or include it right after, which would then immediately disable the loading element as it would not wait for this method to finish. I could add a callback to call `setLoading(false)`, but at that point since I also have an error, I figured I might as well make an undefined Promise.